playground/backend/internal/utils/system_utils.go

```diff
@@ -16,14 +16,59 @@
 package utils
 
 import (
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/logger"
+	"net/http"
+	"os"
+	"path/filepath"
 	"reflect"
 	"runtime"
 	"strings"
 )
 
+const (
+	executableFiles = "executable_files"
+)
+
 // GetFuncName returns the name of the received func
 func GetFuncName(i interface{}) string {
 	fullName := runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
 	splitName := strings.Split(fullName, ".")
 	return splitName[len(splitName)-1]
 }
+
+// GetReadinessFunction returns the function that checks the readiness of the server to process a new code processing request
+func GetReadinessFunction(envs *environment.Environment) func(writer http.ResponseWriter, request *http.Request) {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		if checkNumOfTheParallelJobs(envs.ApplicationEnvs.WorkingDir(), envs.BeamSdkEnvs.NumOfParallelJobs()) {
+			writer.WriteHeader(http.StatusOK)
+		} else {
+			writer.WriteHeader(http.StatusLocked)
+		}
+	}
+}
+
+// checkNumOfTheParallelJobs checks the number of already working code processing.
```

Comment:
of already -> of currently 
code processing -> code executions

Reply:
Done.